tests/test_uncrustify/selector.py

```diff
@@ -0,0 +1,40 @@
+# Utility class to match test filters.
+#
+# * @author  Matthew Woehlke    June 2018
+#
+
+
+# =============================================================================
+class Selector(object):
+    # -------------------------------------------------------------------------
+    def __init__(self, s):
+        class Range(object):
+            pass
+
+        self.ranges = []
+        for p in s.split(','):
+            r = Range()
+            if ':' in p:
+                r.group, p = p.split(':')
+            else:
+                r.group = None
+            if '-' in p:
+                r.lower, r.upper = map(int, p.split('-'))
+            else:
+                r.lower = int(p)
+                r.upper = int(p)
```

Comment:
All of those string to int conversion lack checks, I am not yet sure if that is needed in this case, python will throw exceptions if a non number char is passed right?

Reply:
Yes (`ValueError`). Example:

```
$ tests/run_tests.py --exe $CMAKE_BUILD_DIR/uncrustify -r x
[...]
ValueError: invalid literal for int() with base 10: 'x'
```